doc/source/gotchas.rst

```diff
@@ -242,6 +242,54 @@ Label-based slicing conventions
 Non-monotonic indexes require exact matches
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+If the index of a ``DataFrame`` is monotonically increasing or decreasing, then the bounds
+of a label-based slice can be outside the range of the index, much like slice indexing a 
+normal Python ``list``.
+
```

Comment:
hmm actually what I mean is we need to more references to this section as this is certainly easy to miss. maybe can do that in antother PR though.